framework/src/modules/pos/endpoint.ts

```diff
@@ -245,6 +245,28 @@ export class PoSEndpoint extends BaseEndpoint {
 		return { validators: response };
 	}
 
+	public async getAllValidatorsByStake(
+		ctx: ModuleEndpointContext,
+	): Promise<{ validators: ValidatorAccountEndpoint[] }> {
+		const eligibleValidatorStore = this.stores.get(EligibleValidatorsStore);
+		const validatorSubStore = this.stores.get(ValidatorStore);
+		const response = [];
+
+		const validatorsList = await eligibleValidatorStore.getAll(ctx);
+		for (const { key } of validatorsList) {
+			const [address] = eligibleValidatorStore.splitKey(key);
+			const validatorAccount = await validatorSubStore.get(ctx, address);
+			const validatorAccountJSON = {
+				...codec.toJSON<ValidatorAccountJSON>(validatorStoreSchema, validatorAccount),
+				address: cryptoAddress.getLisk32AddressFromAddress(address),
+				punishmentPeriods: this._calculatePunishmentPeriods(validatorAccount.pomHeights),
+			};
+			response.push(validatorAccountJSON);
+		}
+
+		return { validators: response };
```

Comment:
I think logics are quite similar to the one above `getValidatorsByStake`. I think it would be better to have common function which handles [this](https://github.com/LiskHQ/lisk-sdk/pull/7943/files#diff-a364b32a8b0f42f1ea32864dffc4c65738fbee7577eeafea6c78383b797473f5R256-R265) this part

Reply:
100% agree with @shuse2. The only difference between the 2 methods is how `validatorList` is obtained.

After obtaining the list, those methods could call a private method `_processValidatorList()`

Reply:
Extended `getValidatorsByStake` to return all validator when input `limit` is -1.